system/src/androidLibMain/kotlin/com/splendo/kaluga/system/network/NetworkManager.kt

```diff
@@ -0,0 +1,140 @@
+/*
+ Copyright 2020 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.system.network
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import android.os.Build
+import androidx.annotation.RequiresApi
+import com.splendo.kaluga.logging.debug
+
+sealed class NetworkManager<T>(
+    protected val context: Context
+) : BaseNetworkManager {
+
+    abstract val networkHandler: T
```

Comment:
AFAIK this does not need to be shared because it's not used for anything? Or did I miss it?

Reply:
They are used to register the Callback/BroadcastReceiver

Reply:
Only inside their respective scopes, so it doesn't need to be a field in the parent class.

Reply:
2227902